sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/encoders/ExpressionEncoderSuite.scala

```diff
@@ -242,6 +242,16 @@ class ExpressionEncoderSuite extends SparkFunSuite {
     ExpressionEncoder.tuple(intEnc, ExpressionEncoder.tuple(intEnc, longEnc))
   }
 
+  test("user type with ScalaReflection") {
+    val point = (new ExamplePoint(0.1, 0.2), new ExamplePoint(0.3, 0.4))
+    val schema = ScalaReflection.schemaFor[Tuple2[ExamplePoint, ExamplePoint]]
+      .dataType.asInstanceOf[StructType]
+    val attributeSeq = schema.toAttributes
+    val boundEncoder = encoderFor[Tuple2[ExamplePoint, ExamplePoint]]
+      .resolve(attributeSeq, outers).bind(attributeSeq)
+    assert(boundEncoder.fromRow(boundEncoder.toRow(point)) === point)
+  }
```

Comment:
Just make this `productTest(("UDT", new ExamplePoint(0.1, 0.2)))`.  It always better to reuse existing infrastructure when possible.